beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -364,8 +368,32 @@ impl<E: EthSpec> PeerInfo<E> {
 
     /// Sets an explicit value for the meta data.
     // VISIBILITY: The peer manager is able to adjust the meta_data
-    pub(in crate::peer_manager) fn set_meta_data(&mut self, meta_data: MetaData<E>) {
-        self.meta_data = Some(meta_data)
+    pub(in crate::peer_manager) fn set_meta_data(
+        &mut self,
+        meta_data: MetaData<E>,
+        node_id_opt: Option<NodeId>,
+        spec: &ChainSpec,
+    ) {
+        // If we don't have a node id, we cannot compute the custody duties anyway
+        let Some(node_id) = node_id_opt else {
+            self.meta_data = Some(meta_data);
+            return;
+        };
+
+        // Already set by enr if custody_subnets is non empty
+        if self.custody_subnets.is_empty() {
+            if let Ok(custody_subnet_count) = meta_data.custody_subnet_count() {
```

Comment:
we should verify if the `meta_data.custody_subnet_count` provided is valid (`<= data_column_sidecar_subnet_count`).

Reply:
I've added validation in this PR
https://github.com/sigp/lighthouse/pull/6308